modules/swagger-parser/src/test/java/io/swagger/parser/SwaggerParserTest.java

```diff
@@ -185,9 +185,10 @@ public void testLoadRelativeFileTree_Json() throws Exception {
     public void testLoadExternalNestedDefinitions() throws Exception {
         SwaggerParser parser = new SwaggerParser();
         final Swagger swagger = parser.read("src/test/resources/nested-references/b.yaml");
+
         Map<String, Model> definitions = swagger.getDefinitions();
         assertTrue(definitions.containsKey("x"));
-        assertTrue(!definitions.containsKey("y"));
```

Comment:
Note that `y` lives on, even though it is directly referencing all aspects of `z`